AirCasting/Utils/InAppAlerts.swift

```diff
@@ -0,0 +1,60 @@
+// Created by Lunar on 01/12/2021.
+//
+
+import SwiftUI
+// This struct provides all nedded alerts in case of need. Using them from here, allows you
+// to place multiple instances of them (in one view) and every one of them will be called and show up nicely.
+// Key thing here - is the id.
+// Use them, have fun, ALERT the user 🛎
+struct AlertInfo: Identifiable {
+    enum AlertType {
```

Comment:
Maybe instead of putting app-specific alert types here that would need to be handled inside Views (multiple if-else-fallback scenarios) we can further generalize it by adding `let buttons: [Button]` here and removing `buttonTitle` and the `AlertType`:
```Swift
struct AlertInfo: Identifiable {
    struct Button {
        enum ButtonType { case cancel, ok, destructive }
        let title: String
        let type: ButtonType
        let action: (() -> Void)?
    }
    ...
    let buttons: [Button]
    ...
}
```

and provide needed action blocks inside the static factory methods:

```Swift
static func finishAndSyncAlert(sessionName: String, syncAction: () -> Void) { ... }
```

we can than add an alert factory here too:

```Swift
extension AlertInfo {
   func makeAlert() -> Alert { ... }
}
```

so that inside view we only have to do this:
```Swift
.alert(item: $alert, content: { $0.makeAlert() }
```